lib/nerves_hub_web/live/devices/show.ex

```diff
@@ -502,6 +502,15 @@ defmodule NervesHubWeb.Live.Devices.Show do
     |> assign(:audit_pager, audit_pager)
   end
 
+  defp assign_deployments(%{assigns: %{device: device, product: product}} = socket) do
+    deployments =
+      if device.status == :provisioned,
+        do: Deployments.eligible_deployments(device),
+        else: Deployments.get_deployments_by_product(product)
+
+    assign(socket, deployments: deployments)
+  end
```

Comment:
Tiny nit, don't feel strongly about it, but I do like to burn `if`s to the ground with pattern matching 😈 

```suggestion
  defp assign_deployments(%{assigns: %{device: %{status: :provisioned} = device}} = socket),
    do: assign(socket, deployments: Deployments.eligible_deployments(device))
  
  defp assign_deployments(%{assigns: %{product: product}} = socket),
    do: assign(socket, deployments: Deployments.get_deployments_by_product(product))
```